packages/core/contracts/chainbridge/SourceGovernor.sol

```diff
@@ -16,6 +16,17 @@ contract SourceGovernor is Ownable {
 
     event RelayedGovernanceRequest(uint8 indexed destinationChainId, address indexed to, bytes indexed data);
 
+    /**
+     * @notice Constructor.
+     * @param _finder Address of Finder that this contract uses to locate Bridge.
+     * @param _currentChainId Chain ID for this network. This is configurable by the deployer, rather than
```

Comment:
Lastly, instead of passing the chain ID to the [`SourceGovernor` constructor](https://github.com/mrice32/protocol/blob/5e81b20b13a961cab8f54b89d2e145a946cd9619/packages/core/contracts/chainbridge/SourceGovernor.sol#L24), consider using the assembly `chainid` instruction, to avoid the possibility of mismatches. Note that this suggestion is also applicable to the `BeaconOracle` and its descendant contracts.


@UMAprotocol/eng instead of reading `block.chainid`, which returns a uint256, i still think we should allow the deployer to pass in a uint8 chain ID. This is because the Chainbridge `Bridge.deposit()` method expects a uint8 chain ID to be passed in as the first param. Its possible to lose data when converting from uint256 --> uint8, and the net effect is that some message relays from `Sink` to `Source` might get sent to an unexpected Source contract. This is a problem specifically for the `Oracle` contracts because the `Governor` does not send from `Sink` to `Source`, but for the sake of consistency I recommend leaving this as is.

WDYT?

Reply:
I think keeping these manually specified makes sense for the reasons you outlined. Also for ease of testing. It's harder to do an honest test on a single chain if the chainid is pulled programmatically since both sides would have the same chainid, which makes the test much less realistic.

Reply:
agreed. sgtm